pkg/signing/signing.go

```diff
@@ -134,12 +134,12 @@ func getDataPartitionToSign(fimg *sif.FileImage, dataType sif.Datatype) ([]*sif.
 	return data, nil
 }
 
-// descrToSign determines via argument or interactively which descriptor to sign
-func descrToSign(fimg *sif.FileImage, id uint32, isGroup bool) ([]*sif.Descriptor, error) {
+// descrToSign returns a *sif.Descriptor for every partition to sign determined by argument.
+func descrToSign(fimg *sif.FileImage, id uint32, isGroup, signAll bool) ([]*sif.Descriptor, error) {
 	descr := make([]*sif.Descriptor, 1)
 	var err error
 
-	if id == 0 {
+	if signAll {
```

Comment:
I think this means that if a sif id is specified as well as `-a` then the `-a` has precedence? In which case should we report the conflict on the flags to the user? Just a thought.

Reply:
It should error if any other flag is specified in the cli. _will add that..._